src/RoslynInsertionTool/RoslynInsertionTool/CoreXT.cs

```diff
@@ -39,27 +40,35 @@ internal class CoreXT
         private const string PackagePropsPath = "Directory.Packages.props";
         private const string LegacyPackagePropsPath = "Packages.props";
 
-        private readonly string _defaultConfigOriginal;
+        private readonly string? _defaultConfigOriginal;
 
-        public XDocument ConfigDocument { get; }
+        public XDocument? ConfigDocument { get; }
 
-        private CoreXT(string configOriginalText)
+        private CoreXT(string? configOriginalText)
         {
             _defaultConfigOriginal = configOriginalText;
-            ConfigDocument = XDocument.Parse(configOriginalText, LoadOptions.None);
+            ConfigDocument = configOriginalText is null ? null : XDocument.Parse(configOriginalText, LoadOptions.None);
```

Comment:
Was wondering if this should be something that gets passed in as a parameter to the tool.  We'd update our yaml in main to pass it in (once VS switches over).  New servicing branches would automatically get the value when main merges in, but old ones would continue to use the old behavior

then we could throw on unexpected behavior (e.g. branch was supposed to have a default.config file but we didn't getone)

Reply:
I will punt on this for now. Thanks